src/cmake/helper_methods.cmake

```diff
@@ -0,0 +1,20 @@
+function(add_common_flags target)
+	get_target_property(target_type ${target} TYPE)
+
+	if(MSVC)
+		set_property(TARGET ${target} APPEND_STRING PROPERTY LINK_FLAGS_DEBUG "/DEBUG /INCREMENTAL")
+		set_property(TARGET ${target} APPEND_STRING PROPERTY LINK_FLAGS_RELEASE "/DEBUG /LTCG /INCREMENTAL:NO /OPT:REF")
+		set_property(TARGET ${target} APPEND PROPERTY COMPILE_OPTIONS -DWIN32 -D_WINDOWS -D_WINSOCK_DEPRECATED_NO_WARNINGS -D_CRT_SECURE_NO_WARNINGS)
```

Comment:
I was planning on leaving the warnings in and fixing them later down the road, but I guess that's up to @pond3r

Reply:
Fixing them would be nice. We had to disable the "use safe C library functions" warnings explicitly within the code at my company, since they can be such an annoyance with third party code (and last time I've checked they were only implemented by MSVC, C11 seems to provide them, but IDK about the compiler/C library support). But I'm also will leave the decision up to Tony.

Reply:
I think the right path here is to eventually fix all the warnings and remove the defines which explicitly ignore them so the code just builds clean.

Reply:
OK. Removed ```-D_WINSOCK_DEPRECATED_NO_WARNINGS``` and ```-D_CRT_SECURE_NO_WARNINGS``` from the compile options.